src/software/ai/hl/stp/play/halt_play_test.cpp

```diff
@@ -30,13 +33,13 @@ TEST_F(HaltPlayTest, test_halt_play)
     std::vector<ValidationFunction> terminating_validation_functions = {
         // This will keep the test running for 9.5 seconds to give everything enough
         // time to settle into position and be observed with the Visualizer
-        // TODO: Implement proper validation
-        // https://github.com/UBC-Thunderbots/Software/issues/1971
+
         [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
             while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
             {
                 yield("Timestamp not at 9.5s");
             }
+            robotHalt(world_ptr, yield);
```

Comment:
waiting 9.5 seconds isn't necessary anymore and also it'd be good to check robotHalt over multiple ticks, i.e.
```suggestion
            for (unsigned i = 0; i < 1000; i++)
            {
                robotHalt(world_ptr, yield);
            }
```

Reply:
Fixed